### PR TITLE
JIT: Disable fgMorphCastedBitwiseOp opt for floats

### LINKED ISSUE #61629: Assertion failed 'node->OperIs(GT_ADD, GT_SUB)' in 'Matrix:GetHashCode():int:this' during 'Lowering nodeinfo' (IL size 96)
NGenDump: 
[ngendump.txt](https://github.com/dotnet/runtime/files/7541894/ngendump.txt)

Compile the following with optimizations turned on:

```csharp
using System;
using System.Runtime.InteropServices;

Activator.CreateInstance(typeof(Matrix)).GetHashCode();

[StructLayout(LayoutKind.Sequential)]
public class Matrix : ICloneable
{
    public double Xx;
    public double Yx;
    public double Xy;
    public double Yy;
    public double X0;
    public double Y0;

    public Matrix(double xx, double yx, double xy, double yy,
            double x0, double y0)
    {
        this.Xx = xx; this.Yx = yx; this.Xy = xy;
        this.Yy = yy; this.X0 = x0; this.Y0 = y0;
    }

    public Matrix()
    {
    }

    public bool IsIdentity()
    {
        return (this == new Matrix());
    }

    public void Init(double xx, double yx, double xy, double yy,
              double x0, double y0)
    {
        this.Xx = xx; this.Yx = yx; this.Xy = xy;
        this.Yy = yy; this.X0 = x0; this.Y0 = y0;
    }

    

    public override String ToString()
    {
        String s = String.Format("xx:{0:##0.0#} yx:{1:##0.0#} xy:{2:##0.0#} yy:{3:##0.0#} x0:{4:##0.0#} y0:{5:##0.0#}",
            this.Xx, this.Yx, this.Xy, this.Yy, this.X0, this.Y0);
        return s;
    }

    public static bool operator ==(Matrix lhs, Matrix rhs)
    {
        return (lhs.Xx == rhs.Xx &&
            lhs.Xy == rhs.Xy &&
            lhs.Yx == rhs.Yx &&
            lhs.Yy == rhs.Yy &&
            lhs.X0 == rhs.X0 &&
            lhs.Y0 == rhs.Y0);
    }

    public static bool operator !=(Matrix lhs, Matrix rhs)
    {
        return !(lhs == rhs);
    }



    public override bool Equals(object o)
    {
        if (!(o is Matrix))
            return false;
        else
            return (this == (Matrix)o);
    }

    public override int GetHashCode()
    {
        return (int)this.Xx ^ (int)this.Xx >> 32 ^
            (int)this.Xy ^ (int)this.Xy >> 32 ^
            (int)this.Yx ^ (int)this.Yx >> 32 ^
            (int)this.Yy ^ (int)this.Yy >> 32 ^
            (int)this.X0 ^ (int)this.X0 >> 32 ^
            (int)this.Y0 ^ (int)this.Y0 >> 32;
    }

    public object Clone()
    {
        return this.MemberwiseClone();
    }

}
```

### PR DESCRIPTION
**Pull request by EgorBo**
Fixes https://github.com/dotnet/runtime/issues/61629

### CODE REVIEW

**Review by SingleAccretion — APPROVED** [review]

**Review by jakobbotsch — APPROVED** [review]
LGTM. Some motivation to add float support in Fuzzlyn :)

File: src/tests/JIT/Regression/JitBlue/Runtime_61629/Runtime_61629.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+using System;
+using System.Runtime.CompilerServices;
+
+public class Runtime_61629
+{
+    public static int Main() => 
+        Test(100, 200.0) + Test(Math.PI, Math.PI) - 72;
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    static int Test(double a, double b)
```
Comment (kasperk81) [inline src/tests/JIT/Regression/JitBlue/Runtime_61629/Runtime_61629.cs:13]: does this fail on current main branch?

Comment (EgorBo) [reply src/tests/JIT/Regression/JitBlue/Runtime_61629/Runtime_61629.cs:13]: In Checked builds where we have lots of asserts - yes. For Release one I am not sure

Comment (MichalStrehovsky) [reply src/tests/JIT/Regression/JitBlue/Runtime_61629/Runtime_61629.cs:13]: I was seeing InvalidProgramException getting thrown for the pattern in the issue in release builds. So likely for this one too.

**Pull request merged**